optimum/onnxruntime/modeling_diffusion.py

```diff
@@ -0,0 +1,412 @@
+
+        return cls(
+            *inference_sessions,
+            use_io_binding=use_io_binding,
+            model_save_dir=model_save_dir,
+            tokenizer=sub_models["tokenizer"],
+            scheduler=sub_models["scheduler"],
+            feature_extractor=sub_models.pop("feature_extractor", None),
+            config=config,
+        )
+
+    @classmethod
+    def _from_transformers(
+        cls,
+        model_id: str,
+        config: Optional[str] = None,
+        use_auth_token: Optional[Union[bool, str]] = None,
+        revision: str = "main",
+        force_download: bool = True,
+        cache_dir: Optional[str] = None,
+        subfolder: str = "",
+        local_files_only: bool = False,
+        trust_remote_code: bool = False,
+        provider: str = "CPUExecutionProvider",
+        session_options: Optional[ort.SessionOptions] = None,
+        provider_options: Optional[Dict[str, Any]] = None,
+        use_io_binding: Optional[bool] = None,
+        task: Optional[str] = None,
+    ) -> "ORTStableDiffusionPipeline":
+        if task is None:
+            task = cls._auto_model_to_task(cls.auto_model_class)
+
+        save_dir = TemporaryDirectory()
+        save_dir_path = Path(save_dir.name)
+        model = TasksManager.get_model_from_task(
+            task,
+            model_id,
+            subfolder=subfolder,
+            revision=revision,
+            cache_dir=cache_dir,
+            use_auth_token=use_auth_token,
+            local_files_only=local_files_only,
+            force_download=force_download,
+            trust_remote_code=trust_remote_code,
+        )
+        output_names = [
+            os.path.join(DIFFUSION_MODEL_TEXT_ENCODER_SUBFOLDER, ONNX_WEIGHTS_NAME),
+            os.path.join(DIFFUSION_MODEL_UNET_SUBFOLDER, ONNX_WEIGHTS_NAME),
+            os.path.join(DIFFUSION_MODEL_VAE_ENCODER_SUBFOLDER, ONNX_WEIGHTS_NAME),
+            os.path.join(DIFFUSION_MODEL_VAE_DECODER_SUBFOLDER, ONNX_WEIGHTS_NAME),
+        ]
+        models_and_onnx_configs = get_stable_diffusion_models_for_export(model)
+
+        model.save_config(save_dir_path)
+        model.tokenizer.save_pretrained(save_dir_path.joinpath("tokenizer"))
+        model.scheduler.save_pretrained(save_dir_path.joinpath("scheduler"))
+        if model.feature_extractor is not None:
+            model.feature_extractor.save_pretrained(save_dir_path.joinpath("feature_extractor"))
+
+        export_models(
+            models_and_onnx_configs=models_and_onnx_configs,
+            output_dir=save_dir_path,
+            output_names=output_names,
+        )
+        return cls._from_pretrained(
+            save_dir_path,
+            config=config,
+            provider=provider,
+            session_options=session_options,
+            provider_options=provider_options,
+            use_io_binding=use_io_binding,
+            model_save_dir=save_dir,
+        )
+
+    def to(self, device: Union[torch.device, str, int]):
```

Comment:
Given that this code is duplicate in `modeling_decoder.py`, `modeling_seq2seq.py` and `modeling_diffusion.py`, I think it can make sense to abstract it. But I think it can be done in an other PR.